### PR TITLE
[0383] Added the Course search via Provider for find

### PR DESCRIPTION
**Pull request by defong**
### Context

Course search via Provider for `find`

### Changes proposed in this pull request

Added the Course search via Provider for `find`
Both js and non js version

### Guidance to review
The journey is incomplete as there is
-  nowhere available to go
- validations
- another card for it somewhere

Turn js on and off as there is 2 different user experiences

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Course Search - Initial form by school Javascript version](https://trello.com/c/7uHhLq5c/383-course-search-initial-form-by-school-javascript-version)

**Comment by defong** [comment]
[Course Search Flow - By school](https://trello.com/c/DT8vJSrc/391-course-search-flow-by-school)

File: app/javascript/find/setup-autocomplete.spec.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import setupAutocomplete from './setup-autocomplete'
+
+describe('initACachedProvidersAutocomplete', () => {
```
Comment (darokel) [inline app/javascript/find/setup-autocomplete.spec.js:7]: 👍  for js spec

File: spec/helpers/find/provider_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+require "rails_helper"
+
+describe Find::ProviderHelper, type: :helper do
+  describe "#select_provider_options" do
+    let(:providers) { build_list(:provider, 3) }
+
+    subject { select_provider_options(providers) }
+
+    it "returns select provider options" do
+      expect(subject.map(&:id)).to eql([""] + providers.map { |p| "#{p.provider_name} (#{p.provider_code})" })
```
Comment (darokel) [inline spec/helpers/find/provider_helper_spec.rb:12]: Could we put the expected values behind a meaningful method name so it reads a little better?

**Review by darokel — APPROVED** [review]
Just one little comment but all good

File: spec/helpers/find/provider_helper_spec.rb (hunk 2 above)
Comment (defong) [reply spec/helpers/find/provider_helper_spec.rb:12]: Fixed

File: app/javascript/find/setup-autocomplete.spec.js (hunk 1 above)
Comment (defong) [reply app/javascript/find/setup-autocomplete.spec.js]: Fixed

**Pull request merged**